### DISCUSSION TITLE
v7.0.0

### DISCUSSION THREAD

**Discussion by Taiko2k** [Taiko2k/Tauon#652] [General]
## **The radio UI update**

![Screenshot from 2022-01-10 12-54-12](https://user-images.githubusercontent.com/17271572/148706588-ca499dc1-aa0d-4233-8f20-a17ec7716184.png)

This update features an overhauled internet radio UI. The interface has now become its own layout view rather than a overlay box. You can switch to it using the view switcher shown when clicking MENU.

Notable additions are the support for station thumbnail icons and support for multiple "Lists" (i.e playlists for radios). Also the process for adding and removing stations has been simplified. Another change is that recorded tracks are now imported to a playlist automatically. 

> **Note:** Any previous saved stations will be carried over, but they wont have thumbnails. You'll need to re-save the stations from the station browser to get thumbnails.

> **Note:** The radio record setting has been moved to page 4 of Settings > Function.

Also added **Gensokyo Radio** to the list of supported stations with album art. You can find a list of these stations in Tauon by holding the shift key when viewing the station browser to reveal a secret button.

### Quick tutorial

In the radio layout:
 - Press :heavy_plus_sign: to manually enter a new station from a raw URL.
 - Press :mag: to show the station browser where you can enter a search to a [public database](https://www.radio-browser.info/).
 - Press :star: when shown to save the currently playing station to the current *list*.
 - You can create radio lists using the top panel as you would with a normal playlist. Radio lists are only shown when in the radio layout.
 - You can drag to re-arrange stations in the currently viewed list.

> **Known Issue:** Sometimes starting a station fails and you need to restart Tauon before audio can be played again.

## Other changes:

 - **Jellyfin** server side love/unlove (favourite) is now supported.
   > **Warning:** Initial re-scan of Jellyfin library will cause local love on any Jellyfin tracks to be erased.
 - Export and import of M3U playlists is now supported.
 - Added new generator code `fs` to find exact string segment matches.

___

Upgraders to v7.0.0 will notice a one off donation prompt. I really don't like adding nags/self-promotion, but I need to balance that with what's fair. Feel free to let me know what you think.

<hr /><em>This discussion was created from the release <a href='https://github.com/Taiko2k/TauonMusicBox/releases/tag/v7.0.0'>v7.0.0</a>.</em>

**Comment by talhakerpicci** [comment]
Hey! Thanks for all the efforts. I really enjoy the app! One thing i would like to add is, maybe you could add other donation methods, such as crypto currencies. Therefore people could make one time donations and it would be good for those that wants to stay anonymous. Just my two cents. Thanks for all the effort again! 

Best wishes

**Comment by Solarunit** [comment]
Yay! I have been waiting for the new Radio UI, thanks! 

Re-added all stations and restarted player. Not all stations (from your Picks) have thumbnails

![](https://i.imgur.com/YsIzqdj.png)

Also Listen.moe stations don't have the playing track information at all.
___

>Upgraders to v7.0.0 will notice a one off donation prompt.

I don't mind a one-time donation prompt on new releases, I would like to donate, but can't for various reasons at the moment.
Have you considered other platforms like Patreon or Opencollective?

**Reply by Solarunit** [reply]
I think there should be a Playing radio indicator in the radio list. Maybe a colorful play button? 
Also radio thumbnail on the right is a bit too small?
What is that theme on your screenshot?

**Reply by Taiko2k** [reply]
Ah, the listen.moe not working is an issue with the Flatpak, ill try push out a fix soon.
